go.mod

```diff
@@ -40,23 +40,23 @@ require (
 	github.com/gorilla/mux v1.8.0
 	github.com/rancher/dynamiclistener v0.4.0-rc2
 	github.com/rancher/lasso v0.0.0-20240123150939-7055397d6dfa
-	github.com/rancher/rancher/pkg/apis v0.0.0-20240618132447-2145c8e9d603
-	github.com/rancher/rke v1.5.11-rc1
+	github.com/rancher/rancher/pkg/apis v0.0.0-20240709213639-d72c5157164a
+	github.com/rancher/rke v1.5.11-rc4
 	github.com/rancher/wrangler/v2 v2.1.4
 	github.com/robfig/cron v1.2.0
 	github.com/sirupsen/logrus v1.9.3
 	github.com/stretchr/testify v1.8.4
 	golang.org/x/exp v0.0.0-20230905200255-921286631fa9
-	golang.org/x/text v0.15.0
-	golang.org/x/tools v0.20.0
-	k8s.io/api v0.28.8
-	k8s.io/apimachinery v0.28.8
-	k8s.io/apiserver v0.28.8
+	golang.org/x/text v0.16.0
+	golang.org/x/tools v0.21.1-0.20240508182429-e35e4ccd0d2d
+	k8s.io/api v0.28.9
+	k8s.io/apimachinery v0.28.9
+	k8s.io/apiserver v0.28.9
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/kubernetes v1.28.8
 	k8s.io/pod-security-admission v0.28.6
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
-	sigs.k8s.io/controller-runtime v0.15.3
+	sigs.k8s.io/controller-runtime v0.16.6
```

Comment:
Can you clarify why all of these dependencies were bumped? In particular, I'm concerned about the controller-runtime bump since we do make use of it for some very important webhook functionality: https://github.com/rancher/webhook/blob/7db7696c660bb4c23d0bb6b6ce7b420ab808de3a/pkg/patch/patch.go#L8 .

Reply:
@MbolotSuse So I didn't update them manually but `go mod tidy` picked them up. Looks like rancher v2.8 branch is using this version for controller-runtime 
`sigs.k8s.io/controller-runtime v0.16.6` 

For the k8s components, we replace them with `v0.28.6` in rancher and the require is at `v0.28.9` so my understanding is go picked it up from `require` section. I could add replace directive for k8s components so they match with rancher's go.mod version to keep them consistent. 

For controller-runtime, would we prefer to keep it in line with rancher's version or pinning it to `v0.15.3`?

Reply:
I'd like to avoid a replace directive. I think we can go with this given the integration tests should cover this case and we are above this version on a future WH version. That being said, I'd like to avoid changes like this in the future when we are this close to a release.